daml-assistant/src/DA/Daml/Assistant/Util.hs

```diff
@@ -22,10 +23,10 @@ throwErr msg = throwIO (assistantError msg)
 
 -- | Handle synchronous exceptions by wrapping them in an AssistantError,
 -- add context to any assistant errors that are missing context.
-wrapErrG :: (Text -> SomeException -> AssistantError) -> Text -> IO a -> IO a
-wrapErrG wrapSomeException ctx m = m `catches`
+wrapErrG :: (Text -> SomeException -> AssistantError) -> Maybe Text -> Text -> IO a -> IO a
+wrapErrG wrapSomeException exitMessage ctx m = m `catches`
     [ Handler $ throwIO @IO @ExitCode
```

Comment:
Do we also want to print the exitMessage when we get an actual IO error from the assistant, and not the command? I'd suggest yes, for any eventual uses of wrapErrG that parenthesize errors in the assistant code (e.g. failure to read a file).

Reply:
I'd actually say no, exitMessage is for explicit exits.
Regarding env vars specifically, any errors coming from the assistant itself should not need this context, as they should either provide it themselves, or it isn't related to env vars. i.e your failure to read a file cannot be due to env vars.
If you want information for all errors, I'd say you use the function that adds error context.

Reply:
Remove this from wrapErrG, use `runProcess` instead of `runProcess_` and explicitly print there then exit.